### DISCUSSION TITLE
Someone has a idea How to handle slotname renames / ignore unknown slotnames

### DISCUSSION THREAD

**Discussion by gajanak** [rauc/rauc#1583] [Q&A] [answered]
Hi,

I have a very special situation - where a slotname changes between versions.

In first step this works - because system.conf and bundles matching.
Now I want a simple solution to prevent abort installation of bundle - if bundle contains old slotname - and is not found.
There is no "ignoreSlotUpdateIfMissing" - is there a way to handle this with some kind of hook ?
Or a alias ?

Is it a good Idea to add old and new slotname to system.conf with same config ? This feels like a bad idea ?

Thanks in advantags... ;)

**Comment by jluebbe** [comment]
Ignoring missing slots would go against the concept that the bundle fully describes the intended state of the target system after the update. Accordingly, this is not supported and the abort is intentional. Otherwise, only part of an update would be installed and the target would be in an undefined state.

RAUC's slot names should never need to change, as they are decoupled from device node paths and the names the bootloader uses (which may change from version to version).

**Comment by gajanak** [comment]
Rename Slot was required, because old version of rauc uses slotname as label for mkfs.vfat - and give some problems if longer then 11 chars.
It seems - this is fixed in current releases. (vfat_label_generator)

Can you give me a hint on the second part, is there a big problem if a slot is defined in system.conf twice with different name ?
"Should work..." ? ;)

**Reply by jluebbe** [reply, marked as answer]
> Rename Slot was required, because old version of rauc uses slotname as label for mkfs.vfat - and give some problems if longer then 11 chars. It seems - this is fixed in current releases. (vfat_label_generator)

So your old name was too long?

The fix in https://github.com/rauc/rauc/pull/725 is only needed to work with newer mkfs.vfat versions. So RAUC should still allow long names, they just get truncated in RAUC instead of in mkfs.vfat. I don't see how this would make a rename necessary.

Are you referring to the partition using the label somewhere? If so, please read the note at https://rauc.readthedocs.io/en/latest/integration.html#partitioning-your-device.

> Can you give me a hint on the second part, is there a big problem if a slot is defined in system.conf twice with different name ? "Should work..." ? ;)

While this is not checked during config parsing (yet), it definitely isn't something we can reasonably support. It very likely would lead to problems at least with regards to bootloader interaction. We should probably add a check to protect against such a misconfiguration. :)
